include/hffix.hpp

```diff
@@ -2253,69 +2253,49 @@ class message_reader {
     friend class message_reader_const_iterator;
 
     void init() {
+        char const* b = buffer_; 
 
-        // Skip the version prefix string "8=FIX.4.2" or "8=FIXT.1.1", et cetera.
-        char const* b = buffer_ + 9; // look for the first '\x01'
+        if (b >= buffer_end_) return;
+        if (*b++ != '8') return invalid();
+        if (b >= buffer_end_) return;
+        if (*b++ != '=') return invalid();
 
-        while(true) {
-            if (b >= buffer_end_) {
-                is_complete_ = false;
-                return;
-            }
-            if (*b == '\x01') {
-                prefix_end_ = b;
-                break;
-            }
-            if (b - buffer_ > 11) {
-                invalid();
-                return;
-            }
-            ++b;
-        }
+        // look for the first '\x01'
+        b = (const char*)memchr(b, '\x01', buffer_end_ - b);
```

Comment:
The way this worked before is that we only looked for the `\x01` in the first 11 bytes, because the longest possible version prefix string is `"8=FIXT.1.1"`. I think we should keep that behavior, and not look for `\x01` in the entire buffer.

Reply:
I don't understand why to take the maximum existing length. And if tomorrow there is a new version one character longer? I'll return it as it was ...